### DISCUSSION TITLE
FSRS Java implementation

### DISCUSSION THREAD

**Discussion by zontyp** [open-spaced-repetition/free-spaced-repetition-scheduler#27] [Ideas]
Hi friends,

I am trying to implement FSRS in a java desktop application.
How can I go about using (or creating) the FSRS java implementation.

Please provide your kind guidance.

Best Regards,
Karan

**Comment by L-M-Sherlock** [comment]
I recommend referring to [ts-fsrs](https://github.com/open-spaced-repetition/ts-fsrs).

**Reply by Snoke19** [reply]
@L-M-Sherlock Hello, thank you for your response!
I was wondering if you could guide me on how to approach converting ts-fsrs to a Java version. Should I create an issue in the repository to discuss this, or is it better to contact the authors of the repository directly?
Thank you in advance for your advice!

**Reply by L-M-Sherlock** [reply]
I'm sorry but I don't know Java...

**Reply by Snoke19** [reply]
I’m planning to convert [py-fsrs](https://github.com/open-spaced-repetition/py-fsrs) into a Java project. Would it be okay if I proceed with this implementation? Once it's complete, would it be appropriate to suggest it to you for review or potential publication? also, it would be nice to have some core test cases to be sure that everything is ok

Looking forward to your feedback!

**Reply by L-M-Sherlock** [reply]
I recommend asking the maintainer of py-fsrs.

Btw, here is a Java binding for rs-fsrs.

https://github.com/open-spaced-repetition/rs-fsrs-java

**Reply by Snoke19** [reply]
Thank you for your response!

Regarding the Java binding for rs-fsrs you shared, will this version be consistently updated to align with the stable releases of rs-fsrs?

**Reply by L-M-Sherlock** [reply]
@asukaminato0721 cc

**Reply by asukaminato0721** [reply]
yes, actually this is barely warp the rust version, so as long as rust crate api not change, this will **automatically** get update.

**Reply by asukaminato0721** [reply]
It lacked some apis, but if you read the code, there is no logic there, so you can easily add the api you need there.

**Reply by Snoke19** [reply]
@asukaminato0721 @L-M-Sherlock Thank you both for your answers! 😊
